pkg/body-based-routing/handlers/request.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+	"encoding/json"
+
+	basepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	eppb "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+)
+
+// HandleRequestBody handles request bodies.
+func (s *Server) HandleRequestBody(body *eppb.HttpBody) (*eppb.ProcessingResponse, error) {
+	var data map[string]any
+	if err := json.Unmarshal(body.GetBody(), &data); err != nil {
+		return nil, err
+	}
+
+	modelVal, ok := data["model"]
+	if !ok {
```

Comment:
If we can't find this value in the request body we should likely log, and ideally also track this via a metric at some point in the future. Probably worth filing a follow up issue to cover metrics/observability for this component.

Reply:
Done. Filed issue https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/439